### DISCUSSION TITLE
A Place to Thank the Developer

### DISCUSSION THREAD

**Discussion by pinballelectronica** [LesFerch/WinSetView#66] [General]
I love tool like this that un-f*** WIndows' bad ideas.

Like to share a few more tools that fix Windows problems:

WPD (windows privacy)- needs to be re-run frequently as Windows constantly resets many of it's settings
Everything- replaces Windows crappy search- is super fast and configurable.

Anyhow, thanks for this tool!

**Comment by ptsdon** [comment]
I just wanted to pass along a HUGE thank you for all who were involved putting together this utility. My eyesight isn't so good anymore. Therefore, my preferred view are tiles. I also have a large ICO library that I use to identify folders. 

Windows 10 and 11 are designed so poorly when it comes to respecting the wishes of the user. I have to constantly used CTRL+SHFT+7 to reset my folders. This is extremely annoying when I am teaching an online class. I must have tried to fix this problem dozens of times over the years and nothing worked. But the genus behind this utility is so great that I had to find a way to say it...

THANK YOU!

**Comment by DanteDT** [comment]
This immediately eliminates such a major Windows annoyance that I still cannot believe it.
MANY THANKS!

**Comment by ghost** [comment]
This is a wonderful tool that has saved my sanity. Thank you very much!

**Comment by LJKraus** [comment]
Thank You!  I've been wanting a Detail view that sticks for EVERYTHING for years. Thank You!

**Comment by eightball-unlock** [comment]
Thanks again!

**Comment by Glammerocity** [comment]
Brilliant. It begs the question why can't Microsoft do something this easy? It seems like the issue was created by a MS developer who likes grouping and reseting folder views for his own system, so made the world have it, too. This is why people fear upgrading their OS.

**Comment by rltidd72** [comment]
Thanks very much for this!  Explorer has always been irritating to me, but your app changed all that.  I now get the explorer window the way I want it.  Not microsoft's way.
Thank you! 😂

**Comment by WildByDesign** [comment]
Making Explorer Great Again. 🥇 

Joking aside. Thank you for this incredible program, Les. Explorer became less and less usable and the functionality (as user above me says) became irritating. It took longer to accomplish the job.

My only wish is that more people know about this great program.

**Comment by Arbydoo** [comment]
To everyone on the team, just a huge thank you for your excellent work.

**Comment by That-Ruben** [comment]
The developer is a god for this. I'm ashamed to admit it, but Microsoft's insistence on forcing unwanted garbage into my life has been a constant source of frustration. 

Running this app felt like I was lifting a curse. I feel slightly happier now. 

If anyone knows of an online community where various windows un-frickery is discussed, please let me know. There are several other things I desperately want to stop Windows from doing

**Reply by LesFerch** [reply]
I recommend tenforums.com (for Windows 10) and elevenforum.com (for Windows 11). That's where I first announced WinSetView. There are a lot of good tutorials and discussions there for just about everything that can be tweaked in Windows. Two other tools that you may find useful are [Winaero Tweaker](https://winaero.com/winaero-tweaker/) and [Ultimate Windows Tweaker](https://www.thewindowsclub.com/ultimate-windows-tweaker-4-windows-10).

**Comment by kf-techwrite-git** [comment]
I'm without words... I want to bear your children. In fact, many women should bear as many of your children as possible. In so this tribe would save our world. 
Srsly, my sincerest thanks for this work; I wish everyone knew about it. Thank you thank you thank you!

**Reply by LesFerch** [reply]
You're most welcome and, uh, thanks! 😳

**Reply by JoanEliot** [reply]
> I want to bear your children. In fact, many women should bear as many of your children as possible. In so this tribe would save our world.
 Men and women both. 
 Such a relief finally to have found this tool.

**Comment by RainGater** [comment]
In a word, WOW!  

WinSetView is a wonderful utility and can't thank enough.  Windows explorer is heavily used and the group by and the columns, width, settings, etc. keep changing everytime Microsoft does some trial and error and WinSetView takes care of all these issues and much more!

No other way to put it other than to say, Just Love It!

EDIT: When I open or save file, the dialog doesn't really honor the column width that I set for the default view.  Is it by design or is there anyway to force the column width of the open/save dialog to display the same way as the default explorer view?

**Comment by LesFerch** [comment]
> When I open or save file, the dialog doesn't really honor the column width that I set for the default view. Is it by design or is there anyway to force the column width of the open/save dialog to display the same way as the default explorer view?

The file dialogs views should honor your settings, including column widths. The usual reason for not getting the expected result is that the app was open at the time Submit was clicked in WinSetView. Some apps minimize to the System tray when you click X. They have to be fully closed (e.g. right-click their system tray item and select Quit or Exit).

If you're seeing this issue in all apps, then something else is up. What apps have you checked?

**Comment by RainGater** [comment]
You are correct. Notepad++ and ShareX were OPEN when I submitted the
settings from WinSetView.  Now, I closed them and when I opened it, it was
not honoring the column width.  Then, I re-ran the setup and now it works
fine.  I guess all the apps need to be closed before running your app?  If
that is the case, I guess I will have to run it after I reboot my computer.

**Reply by LesFerch** [reply]
The apps need to be closed before clicking Submit. Rebooting (or logging out and in) before running WinSetView is a good idea unless you have apps set to automatically load at startup.

**Comment by RainGater** [comment]
Thank you for your response. Yes, I have apps set to auto run at startup.
That's fine as I closed the auto start up apps and re-ran the setup.

Thanks again for the wonderful utility and it's an invaluable tool, imho.

**Comment by gretchenplant** [comment]
Thank you so much for this tool! It has solved about 15 issues I've been tasked with finding solutions for for my team at work (and also all the personal Win 11 machines in my friends and family) all in one easy to use, and exceptionally documented program. You're a true hero!!!

**Reply by LesFerch** [reply]
Thank you! Your message really made my day. 🙂

**Comment by Osupuko3745** [comment]
You are a superstar! This tool has made my life absolutely livable especially on windows 11 which unlike windows 10 somehow does not apply views to all folders. Where to buy you coffee? RIP Puddles.

**Reply by LesFerch** [reply]
So nice to hear that! Thanks for the coffee offer, but no need. Glad to help!
Here's a picture of Puddles:
![image](https://github.com/user-attachments/assets/94a8d4d9-d2fc-4799-ba8a-0a2529de8c12)

**Reply by Osupuko3745** [reply]
Cute lil guy!... Thank you so much once again.

**Comment by Hornblower409** [comment]
Thank you so much. Just FYI if you track referrals. I found [the link to it] [here](https://answers.microsoft.com/en-us/windows/forum/all/completely-disable-file-grouping-always-everywhere/ac31a227-f585-4b0a-ab2e-a557828eaec5).

**Comment by davekingSHS** [comment]
Thank you so much for fixing this major annoyance. You are a life-saver. Whoever on the Windows team thought that grouping files was a good idea should be taken out back and... I'm kidding, of course.

**Comment by GambotTheta** [comment]
Thank you for sharing such an amazing tool with the world. You're breathtaking!

**Comment by beepinboop** [comment]
Thank you very, very much for the tool! I hope you're having a nice week!

**Comment by Gibdos** [comment]
Just came across this great little tool. **Thank you very much** for fixing Microsofts idiotic decisions and giving me at least a little control back to my PC.

**Comment by Laidbak300** [comment]
This is a lifesaver. I've been battling the differing M$ views since it became lots harder in Windows Vista/7 to set a single view everywhere for everything. Windows 11 is so bad that I've been struggling with getting a new laptop set up full time for 3 weeks now; it's wasted all my summer spare time and vacation. I might as well have moved to Linux--and will next time. I am not going through this hell for "Windows 12". I've always been afraid of the Linux learning curve but now I'm taking so much time tweaking Windows--and M$ has made it so much harder to work in the registry--that the Linux learning curve can't be any worse at this point. If I'd only known, I'd have made a different purchase when I bought my new (Windows) laptop. But at least your utility will help me stumble through the next couple of years.

**Comment by Gdrals** [comment]
Thank you for this app. I was tearing my hair out, trying to organise stuff on my phone, because it reverts to the god-damned "Side by side" view for EVERY folder, EVERY time I open another folder, even when I put it to Detailed view.
Then I found this wonderful app. And thanks to the discussion of issues, I was able to find out how to make it work on mobile devices too (Folder View Options > Check "Use General Items view for connected devices".

So thank you! 😊

**Reply by LesFerch** [reply]
You're very welcome!

Just for reference, you can find the setting in the manual by searching for "phone". Here's a direct link to relevant section:

https://lesferch.github.io/WinSetView/doc/v3/Manual.html#use-general-items-view-for-connected-devices

**Comment by That-Ruben** [comment]
Windows is just so broken if you look at it closely. It has so many settings that just don't work at all. Like the "optimize this folder for" in the Customize tab of folder properties in Explorer. Windows put all my sound file folders into an "Artist, Album, Genre" format, and utterly refused to apply the Customize settings. 

That's how I found my way back to WinSetView once again

**Reply by LesFerch** [reply]
The "optimize this folder for" option with the Windows 11 Explorer only works for one folder at a time (i.e. you have to set every subfolder individually) and it only works for local disk NTFS volumes. The option "Also apply this template to all subfolders" has been broken since early 2022. Switching to the Windows 10 Explorer resolves those issues, but I sure wish they'd fix the problem in the new Explorer.

Also note that, for local disk NTFS volumes, you can use my [SetFolderType](https://lesferch.github.io/SetFolderType/) tool to set the folder type for a whole folder tree. This will work with the Windows 11 Explorer.

**Comment by davekingSHS** [comment]
Ugh, I set most settings to "content", and file save dialogs still show "details" (would be nice if developers kept these dialogs updated). Windows really sucks sometimes.
I use a replacement file manager for the most part, Directory Opus. It's great at sorting folders and files the way you want them, and it has a dual tree view, and dual file lists.

**Reply by LesFerch** [reply]
I just set Contents as my Global view setting and all of my open/save dialogs were in content view. As per the readme:

> **Note**: For best results, close all open applications before running WinSetView. Open applications can prevent open/save dialog views from being updated. Apps that minimize to the System Tray when “closed”, such as Discord, Steam, and qBittorrent, must be fully closed (e.g. right-click the app’s System Tray icon and select Exit).

If that doesn't explain it, please let me know which apps were still in Details view and I will try to replicate.

**Reply by davekingSHS** [reply]
Sorry, I missed the readme. Followed it, and all is good now.

**Comment by veenhond2013** [comment]
What a genius little program. It just feels so pleasant to tweak this and see the blue shellscreen overwriting Windows.  This is peak computing right there. Thanks so much

**Comment by DarkFox218** [comment]
Thank you developer, I can not believe windows 11 still has this issue, your app is a blessing upon this cursed operating system

**Comment by Jeaudoir** [comment]
This app is incredible. Where has it been all my life?! Les, is there a PayPal address (or something) where I can send you some money? Not a ton as I'm quite broke, but I would definitely love to send you some. You've saved me who-knows-how-many hours of frustration.

**Reply by LesFerch** [reply]
Thank you for the offer, but my apps are truly 100% free. Enjoy.

**Comment by wilsonletras** [comment]
Good stuff, works like a charm! Thank you!

**Comment by kevinruffus** [comment]
Thank you! This saved me countless hours of searches and manual tweaks/edits.

**Comment by rudolf-weissenegger** [comment]
Altough I (re)switched to Linux two months ago, I still need to use Windows for a programming project.
Since the last update to Win11 25H2 my manual settings for the "Group by" in File Explorer were only applied to the actual folder and resetted every time I closed/reopened File Explorer. This sincerly drove me crazy!

So **thank you very much** for making my life a bit easier, every time I need to run Windows.

**Comment by WizardGrimm** [comment]
Where have all of you been all my life?  At least since the switch in Windows 10 of the Download folder to an independent default-sort-by-Group folder. One of the most annoying little things Microsoft ever inflected on users.  Not only has @WinSetView solved that grouping problem, but this crazy little app performs huge fixes on how my folders are organized.  "_Many thanks_" is only a limited expression of my sincerest gratitude. Thank you; thank you; thank you!

**Comment by phonecardmike** [comment]
I want to say **THANK YOU SO MUCH FOR THIS PROGRAM!** I have been searching for this for months.  So far it works great and will save me a ton of time.  You rock!!

**Comment by Stef-anL** [comment]
Great! Thank you!
Stefan

**Comment by CFLip2X** [comment]
I just want to say thank you this has been great been using it for years. One thing I didn't know it did till recently was when I make shortcuts to a video file (and probably music but haven't checked) it adds the meta data and rating. It makes it easier to organize videos that are in different locations but can be in different folder locations.

**Reply by LesFerch** [reply]
Thank you, but just to clarify, WinSetView is not providing that feature. I assume it's just something Explorer does, but I'd have to do some research to confirm that.

**Reply by CFLip2X** [reply]
Well then it might just be a happy coincidence. I just installed it on a spare laptop that never had the program installed and when I select the columns that I want visible it appears to copy that metadata into the .lnk shortcut details view. I need to make sure info I want needs to be at least checked off in the Add to Right-Click Mean in the columns are under the Global (Details) for it copy the information. If change the rating of the original file the shortcut still works though the rating does not change which is totally fine with me.

**Comment by kim2048** [comment]
Awesome, I love it ❤️ 

Especially the "coder ui", reminds me of the ui of https://xnormal.net/ :)

Only thing missing is an open source version of [size in bytes](https://www.smartftp.com/en-us/sizeinbytes) (I really hate mixing byte units in listings)

**Reply by LesFerch** [reply]
Thank you. 😊

Microsoft announced a change coming to Explorer to show files in units appropriate to their size (KB, MB, GB) but my Explorer still shows everything in KB. Are you seeing a mix of units? Maybe you have that update?

Anyhow, WinSetView only handles settings that can be done via the registry. It has no resident code and I'm pretty sure it would take a hook type of extension to modify Explorer's size column display.

WindHawk is a possible route for a free open source mod to display everything in bytes. Maybe try contacting the developer of the [Better file sizes in Explorer details](https://windhawk.net/mods/explorer-details-better-file-sizes) mod and see if they can add an option to always show sizes in bytes.

**Reply by kim2048** [reply]
Hi, you are right, since 11 everything is in KB - but as a coder, only bytes matter ;)

I wasn't aware WindHawk had such a plugin, I will have a look at implementation and see what I can do. Thanks for the hint!

**Comment by JeFtFotF** [comment]
Mr. Ferch,
Thank you for your great work with WinSetView that helps me and others stop Windows Explorer (Win11 Pro) from driving us to deep distraction with its stubborn "Group By" functions. I recently reset Windows after an ugly crash, then the non-None aspect of the "Group By" function showed up uninvited. I have wiped this away before but this time it was disturbingly persistent. I found your solution on several sites (e.g., Windows Eleven Forum, SuperUser, etc.) via search. I downloaded your files and solved that nonsense. Thanks again!
Sincerely,
JeFtFotF

**Comment by FVolral** [comment]
Dude... I LOVE YOU !!!
